filepathfilter/filepathfilter.go

```diff
@@ -16,18 +16,35 @@ type Pattern interface {
 }
 
 type Filter struct {
-	include []Pattern
-	exclude []Pattern
+	include      []Pattern
+	exclude      []Pattern
+	defaultValue bool
 }
 
-func NewFromPatterns(include, exclude []Pattern) *Filter {
-	return &Filter{include: include, exclude: exclude}
+type options struct {
+	defaultValue bool
 }
 
-func New(include, exclude []string) *Filter {
+type option func(*options)
+
+func DefaultValue(val bool) option {
```

Comment:
My only hesitation here is that `DefaultValue` is something of a generic phrase and, while I have no good alternatives to suggest, I wondered if we could add a comment describing it's function, at least.

I think my hesitation stems from the fact that, IIUC, this isn't so much a default in the sense of "if no patterns match, then do X" but rather "if no **_include_** patterns are **_defined_**, then do X".  So it feels more specific than a generic default ... maybe there's a way to express that in a different name, like `NoIncludeDefaultValue`?  Sheesh, naming things is hard.  :-(

Reply:
True, we only consider include patterns, but this is an optimization: if we know that the filter has only excludes left, the only thing we can return is `false` or the default value.  If the default value is `false`, then we need not consider the excludes at all, since we know what the return value is.  The optimization isn't required for correctness, but it's clear to me that we need multiple additional comments here, since this is tricky to reason about.

Reply:
I've added a comment here explaining what `DefaultValue` does and explained the logic here much better.

Reply:
Thanks, and I think I noticed the logic changed too to make it more clearer as well!  ❤️